test/e2e/main_test.go

```diff
@@ -77,11 +56,31 @@ func TestMain(m *testing.M) {
 
 		if err := framework.Teardown(); err != nil {
 			log.Printf("failed to teardown framework: %v\n", err)
-			code = 1
+			exitCode = 1
 		}
 
-		os.Exit(code)
+		os.Exit(exitCode)
 	}()
 
-	code = m.Run()
+	exitCode = m.Run()
+
+	// Check if Prometheus Operator ever restarted.
+	restarts, err := framework.GetRestartCount(framework.Namespace.Name, framework.OperatorPod.Name)
+	if err != nil {
+		log.Printf("failed to retrieve restart count of Prometheus Operator pod: %v", err)
+		exitCode = 1
+	}
+	if len(restarts) != 1 {
+		log.Printf("expected to have 1 container but got %v", len(restarts))
```

Comment:
consider using `%d` to format integers

Reply:
Using `%v` uses the default format of the `Stringer` interface, thereby a base10, right?

Reply:
Sure, but `%d` is designed for integers. Why should we be less specific?